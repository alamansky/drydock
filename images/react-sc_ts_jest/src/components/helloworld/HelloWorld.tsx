import React from 'react';
import styled from 'styled-components';

const Title = styled.h1`
	font-size: 1.5em;
	text-align: center;
`;

interface HelloWorldProps {
	text: string;
}

const HelloWorld = ({ text = 'hello world' }: HelloWorldProps) => {
	return <Title>{text}</Title>;
};

export default HelloWorld;
