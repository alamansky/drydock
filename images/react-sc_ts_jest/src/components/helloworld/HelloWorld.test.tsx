/* eslint-env jest */

import React from 'react';
import HelloWorld from './HelloWorld';
import { shallow } from 'enzyme';

const text = 'hello';

describe('<HelloWorld/>', () => {
	it('renders correctly', () => {
		const wrapper: any = shallow(<HelloWorld text={text} />);
		console.log(wrapper.text());
		expect(wrapper.text()).toBe(text);
		expect(wrapper).toMatchSnapshot();
		return;
	});
});
