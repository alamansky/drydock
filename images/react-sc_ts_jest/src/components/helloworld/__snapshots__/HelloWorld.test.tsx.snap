// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HelloWorld/> renders correctly 1`] = `
<styled.h1>
  hello
</styled.h1>
`;
